src/test/java/uk/gov/pay/directdebit/payments/dao/PaymentDaoIT.java

```diff
@@ -209,11 +210,105 @@ public void shouldReturnEmptyWhenProviderIdMatchesButProviderDoesNotMatch() {
+        var expectedProviderId = SandboxPaymentId.valueOf("aProviderId");
+
+        testGatewayAccount = aGatewayAccountFixture()
+                .withPaymentProvider(SANDBOX)
+                .withOrganisation(null)
+                .insert(testContext.getJdbi());
+
+        testMandate = MandateFixture.aMandateFixture()
+                .withGatewayAccountFixture(testGatewayAccount)
+                .insert(testContext.getJdbi());
+
+        testPayment
+                .withMandateFixture(testMandate)
+                .withPaymentProviderId(expectedProviderId)
+                .insert(testContext.getJdbi());
+
+        Payment payment = paymentDao.findPaymentByProviderId(SANDBOX, expectedProviderId).get();
+
+        assertThat(payment.getId(), is(testPayment.getId()));
+        assertThat(payment.getExternalId(), is(testPayment.getExternalId()));
+        assertThat(payment.getProviderId().get(), is(expectedProviderId));
+    }
+    
+    @Test
+    public void shouldNotGetAPaymentByProviderIdAndNoOrganisationIdIfPaymentProviderIdDoesNotMatch() {
+        var expectedProviderId = SandboxPaymentId.valueOf("aProviderId");
+
+        testGatewayAccount = aGatewayAccountFixture()
+                .withPaymentProvider(SANDBOX)
+                .withOrganisation(null)
+                .insert(testContext.getJdbi());
+
+        testMandate = MandateFixture.aMandateFixture()
+                .withGatewayAccountFixture(testGatewayAccount)
+                .insert(testContext.getJdbi());
+
+        testPayment
+                .withMandateFixture(testMandate)
+                .withPaymentProviderId(expectedProviderId)
+                .insert(testContext.getJdbi());
+
+        Optional<Payment> payment = paymentDao.findPaymentByProviderId(SANDBOX, SandboxPaymentId.valueOf("differentPaymentId"));
+
+        assertThat(payment, is(Optional.empty()));
+    }
+
+    @Test
+    public void shouldNotGetAPaymentByProviderIdAndNoOrganisationIdIfProviderDoesNotMatch() {
+        var expectedProviderId = SandboxPaymentId.valueOf("aProviderId");
+
+        testGatewayAccount = aGatewayAccountFixture()
+                .withPaymentProvider(SANDBOX)
+                .withOrganisation(null)
+                .insert(testContext.getJdbi());
+
+        testMandate = MandateFixture.aMandateFixture()
+                .withGatewayAccountFixture(testGatewayAccount)
+                .insert(testContext.getJdbi());
+
+        testPayment
+                .withMandateFixture(testMandate)
+                .withPaymentProviderId(expectedProviderId)
+                .insert(testContext.getJdbi());
+
+        Optional<Payment> payment = paymentDao.findPaymentByProviderId(GOCARDLESS, expectedProviderId);
+
+        assertThat(payment, is(Optional.empty()));
+    }
+
+    @Test
+    public void shouldNotGetAPaymentByProviderIdAndNoOrganisationIdIfOrganisationDoesNotMatch() {
+        var expectedProviderId = SandboxPaymentId.valueOf("aProviderId");
```

Comment:
Didn't quite get the intent of this test from the name. Am I right in thinking it shouldn't return a payment because when searching sandbox using payment provider Id the organisation must be null? If so, the `DoesNotMatch` confused me slightly because I was looking for the miss-match between what we set it to and what we search with, maybe rename it `shouldNotGetAPaymentByProviderIdAndNoOrganisationIdIfOrganisationIsNotNull`?

Reply:
What it’s doing is inserting a payment with a provider ID and an organisation ID and then searching for that same provider ID but no organisation and checking it doesn’t get the payment. Your name is better though, so I’ll update it.